### PR TITLE
Chrome: Hide several Sidebar Panels if the CPT doesn't support it

### LINKED ISSUE #3989: Many blocks appears in sidebar for CPTs that do not support certain features
## Issue Overview

I created a custom post type for testing. Although the custom post type did not support featured images, excerpts, tags, categories, comments, or pingbacks, the blocks for these appeared in the Gutenberg sidebar.

## Steps to Reproduce (for bugs)

1. Register a custom post type without support for featured images, excerpts, tags, categories, comments, or pingbacks. Mine has the following `$args`:

```
$args = array(
	'labels'            => $labels,
	'public'            => true,
	'hierarchical'      => false,
	'show_ui'           => true,
	'show_in_nav_menus' => true,
	'supports'          => array( 'title', 'editor' ),
	'has_archive'       => true,
	'rewrite'           => true,
	'query_var'         => true,
	'show_in_rest'      => true,
	'rest_base'         => 'example',
	'rest_controller_class' => 'WP_REST_Posts_Controller',
);
```
2. Open a new post of the `example` custom post type for editing using Gutenberg.

[Safari/Chrome, macOS High Sierra, WordPress 4.9.1, Gutenberg 1.9.1]

## Expected Behavior
The blocks for featured images, excerpts, tags, categories, comments, or pingbacks do not appear in the Gutenberg sidebar.

## Current Behavior
The blocks for featured images, excerpts, tags, categories, comments, and pingbacks appear in the Gutenberg sidebar. However, WordPress does not reflect settings added via these blocks in Gutenberg on the front end as the post type does not support them.

## Possible Solution
Gutenberg should only show these blocks in the sidebar if the current post type supports them, which necessitates access to `post_type_supports()` data for any given custom post type.

## Screenshots / Video
Gutenberg:
screen shot 2017-12-13 at 11 18 09 pm

Classic Editor:
screen shot 2017-12-13 at 11 12 24 pm

## Related Issues and/or PRs
Semi-related: https://github.com/WordPress/gutenberg/issues/2523

## Todos
- [ ] Tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by youknowriad**
closes #3989

This PR hides the featured image panel on CPT not supporting it.

**Testing instructions**

- Register a CPT not supporting the "thumbnail", "comments", "trackbacks", "excerpt", "revisions", "author" and "post-formats"
- Create a new post for this CPT
- The corresponding panel should not show up.

### CODE REVIEW

File: editor/components/post-type-support-check/index.js [hunk 1]
```diff
@@ -16,7 +16,9 @@ import { compose } from '@wordpress/element';
 import { getCurrentPostType } from '../../store/selectors';
 
 function PostTypeSupportCheck( { postType, children, supportKey } ) {
```
Comment (jorgefilipecosta) [inline editor/components/post-type-support-check/index.js]: Maybe the supportKey should be named supportKeys given that we accept more than one.

**Review by jorgefilipecosta — APPROVED** [review]
I tested the changes and things look as expected. Nice work 👍

File: editor/components/post-type-support-check/index.js [hunk 2]
```diff
@@ -15,8 +15,10 @@ import { compose } from '@wordpress/element';
  */
 import { getCurrentPostType } from '../../store/selectors';
 
-function PostTypeSupportCheck( { postType, children, supportKey } ) {
-	const isSupported = get( postType, [ 'data', 'supports', supportKey ], false );
+function PostTypeSupportCheck( { postType, children, supportKeys } ) {
+	const isSupported = some(
```
Comment (aduth) [inline editor/components/post-type-support-check/index.js]: Should this be `every`? I could imagine a need for both cases, i.e. a component that requires post type support for both X _and_ Y, but alternatively X _or_ Y.

Comment (youknowriad) [reply editor/components/post-type-support-check/index.js]: I hesitated on adding an "operator". For now we only have the "or" use-case.

**Pull request merged**